web/film-club-client/src/App.js

```diff
@@ -1,103 +1,33 @@
-import React, { useState, useContext, useEffect } from 'react';
-import { Box, Button, Collapsible, Card, Grid, Grommet, Heading, InfiniteScroll, Layer, ResponsiveContext, Sidebar, Text, grommet } from 'grommet';
-import { FormClose, Menu, Search } from 'grommet-icons';
-import { deepMerge } from 'grommet/utils'
+import React, { useState, useEffect } from 'react';
+import { Route, Switch } from 'react-router-dom';
 
 // Import Custom Components
-import PosterCard from './components/PosterCard'
-import Header from './components/Header'
+import Home from './components/Home'
+import FilmDetailPage from './components/FilmDetailPage'
 
-// // Grommet Responsive Grid Configuration
-// Declare custom Breakpoints for screen size
-const customBreakpoints = deepMerge(grommet, {
-  global: {
-    breakpoints: {
-      small: {
-        value: 600,
-      },
-      medium: {
-        value: 900,
-      },
-      large: {
-        value: 2000,
-      },
-    },
-  },
-})
-
-// Grid Column Layout
-const columns = {
-  small: ['auto'],
-  medium: ['50%', '50%'],
-  large: ['50%', '50%'],
-}
-
-// Grid Row Layout
-const rows = {
-  small: ['auto'],
-  medium: ['auto'],
-  large: ['auto'],
-}
-
-// Responsive Grid Logic
-const Responsive = ({ children, ...props }) => {
-  const size = useContext(ResponsiveContext)
-  let rowsVal = rows
-  if (rows) {
-    if (rows[size]) {
-      rowsVal = rows[size]
-    }
-  }
-
-  let columnsVal = columns
-  if (columns) {
-    if (columns[size]) {
-      columnsVal = columns[size]
-    }
-  }
-
-  return (
-    <Grid
-      rows={ !rowsVal ? size : rowsVal }
-      columns={ !columnsVal ? size : columnsVal }
-      {...props}
-    >
-      {children}
-    </Grid>
-  )
-}
 
 function App() {
-  const [showSidebar, setShowSidebar] = useState(false)
+  // const [showSidebar, setShowSidebar] = useState(false)
   const [movies, setMovies] = useState([])
+  const [film, setFilm] = useState({})
 
-  // Get first batch of film data
+  // Get all film data and concat to 'movies' array for Infinite Scroll use
   const getUpcomingMovies = async () => {
-    const page = (movies.length / 20) + 1
-    const response = await fetch(`/films/upcoming?page=${page}`)
-    const data = await response.json()
-    setMovies(movies.concat(data.films))
+      const page = (movies.length / 20) + 1
+      const response = await fetch(`/films/upcoming?page=${page}`)
+      const data = await response.json()
+      setMovies(movies.concat(data.films))
   }
```

Comment:
This could probably move into the `Home` component along with the `const [movies, setMovies] = useState([])` line.